src/blocks/subscribe/index.php

```diff
@@ -490,12 +490,16 @@ function process_form() {
 		$metadata['status'] = 'pending';
 	}
 
+	$registered_user = false;
 	if ( ! \is_user_logged_in() && \class_exists( '\Newspack\Reader_Activation' ) && \Newspack\Reader_Activation::is_enabled() ) {
 		$metadata = array_merge( $metadata, [ 'registration_method' => 'newsletters-subscription' ] );
 		if ( $popup_id ) {
 			$metadata['registration_method'] = 'newsletters-subscription-popup';
 		}
-		\Newspack\Reader_Activation::register_reader( $email, $name, true, $metadata );
+		$registered_user = \Newspack\Reader_Activation::register_reader( $email, $name, true, $metadata );
+		if ( $registered_user ) {
```

Comment:
If `register_reader()` returns a `WP_Error` it will be truthy. This check needs to change.